plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/PipelineArgumentsTab.java

```diff
@@ -439,6 +439,7 @@ private boolean validateRequiredGroups(MissingRequiredProperties validationFailu
     if (!validationFailures.getMissingGroups().isEmpty()) {
       Map.Entry<String, Set<PipelineOptionsProperty>> missingGroupEntry =
           Iterables.getFirst(validationFailures.getMissingGroups().entrySet(), null);
+      Preconditions.checkNotNull(missingGroupEntry); // should be impossible
```

Comment:
Eclipse's nullness checker is tripped up as `Iterables.getFirst()` is annotated with a `@Nullable` return.  Note that this should never happen because of the check on line 439.

An alternative might be:
```java
Map.Entry<String, Set<PipelineOptionsProperty>> missingGroupEntry =
           Iterables.getFirst(validationFailures.getMissingGroups().entrySet(), null);
if (missingGroupEntry != null) {       
```